File: Go/Two Sum II - Input array is sorted.go
// 4ms, 95.14%; 3MB, 100.00%
func twoSum(numbers []int, target int) []int {
	i, j := 0, len(numbers)-1
	for {
		sum := numbers[i] + numbers[j]
		if sum < target {
			i++
		} else if sum > target {
			j--
		} else {
			break
		}
	}
	return []int{i + 1, j + 1}
}

File: Java/Two Sum II - Input array is sorted.java
// 1ms, 48.02%; 40.1MB, 5.22%
class Solution {
    public static int[] twoSum(int[] numbers, int target) {
        int[] idx = { 1, numbers.length };
        while (true) {
            int tmp = numbers[idx[0] - 1] + numbers[idx[1] - 1];
            if (tmp < target) {
                ++idx[0];
            } else if (tmp > target) {
                --idx[1];
            } else {
                break;
            }
        }
        return idx;
    }
}
